### ISSUE TITLE
Hands Gesture Recognition

### ISSUE THREAD

**Issue by abhisheks008** [abhisheks008/DL-Simplified#17]
### Deep Learning Simplified Repository (Proposing new issue)
:red_circle: **Project Title** : Hands Gesture recognition</br>
:red_circle: **Aim** : Create a model which will recognize the gestures made by the hands.</br>
:red_circle: **Dataset** : https://www.kaggle.com/datasets/kapitanov/hagrid</br>
:red_circle: **Approach** : Try to use 3-4 algorithms to implement the models and compare all the algorithms to find out the best fitted algorithm for the model by checking the accuracy scores. Also do not forget to do a exploratory data analysis before creating any model.

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : 
- GitHub Profile Link : 
- Email ID :
- Participant ID (if applicable):
- Approach for this Project :
- What is your participant role? (Mention the Open Source program)

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by ArthrowAbstract** [comment]
Full Name : Devanshu Sinha
GitHub Profile Link : https://github.com/ArthrowAbstract
Email ID : sinhadevanshu2002@gmail.com
What is your participant role? : Contributor in SSOC

**Comment by LahariBoni** [comment]
Issue Assigned to you @ArthrowAbstract, Go ahead

**Comment by abhisheks008** [comment]
@ArthrowAbstract updates please.

**Comment by abhisheks008** [comment]
Updates please @ArthrowAbstract

**Comment by Soumadeep03** [comment]
Full Name- Soumadeep Maji
GitHub Profile Link- https://github.com/Soumadeep03
Email ID- majisoumadeep2003@gmail.com
Want to Contribute to this issue

**Comment by abhisheks008** [comment]
Can you elaborate your approach for solving this issue?
@Soumadeep03

**Comment by Anushkabajpai11** [comment]
Full Name- Anushka Bajpai
GitHub Profile Link- https://github.com/Anushkabajpai11
Email ID- Anushkabaj1111@gmail.com
Want to Contribute to this issue

**Comment by abhisheks008** [comment]
Can you share the approach you want to implement here in this project?
@Anushkabajpai11

**Comment by harshitaaa8** [comment]
Full Name- Harshita
GitHub Profile Link- https://github.com/harshitaaa8
Email ID- harshi3925@gmail.com
Want to Contribute to this issue under SSOC as a beginner.
Will use CNN approach and MediaPipe library.

**Comment by abhisheks008** [comment]
Issue assigned to you @harshitaaa8

**Comment by SaiBhaskar0987** [comment]
Full name : Sai Bhaskar Kandula
GitHub Profile Link: https://github.com/SaiBhaskar0987
Email ID : saibhaskark05@gmail.com
Participant ID (if applicable):
Approach for this Project : 

First of all to create an effective hand gesture recognition model, i will start by collecting a comprehensive dataset of hand gestures and perform exploratory data analysis to understand image diversity and distribution, Preprocess this data by normalizing, resizing, and augmenting the images, and encode labels numerically. Implement and compare various machine learning algorithms such as Convolutional Neural Networks (CNN), Support Vector Machines (SVM), Random Forests, and K-Nearest Neighbors (KNN) to identify which performs best using metrics like accuracy and F1-Score. Leverage frameworks such as TensorFlow for CNNs and Scikit-learn for traditional algorithms, optimizing for the highest accuracy and robustness to select the most suitable model for deployment in practical applications.

What is your participant role: GSSOC 2024 Contributor

@abhisheks008 assign me this issue

**Comment by abhisheks008** [comment]
Assigned to you @SaiBhaskar0987

**Comment by abhisheks008** [comment]
> Dear Sir, @abhisheks008 I hope this message finds you well. I am writing to discuss the hand gesture recognition project I have been assigned to. I noticed that the dataset provided for this project is in the form of Kaggle dataset, JSON files. I am keen to ensure that my work aligns with the project’s guidelines and objectives. Could you please clarify if it is mandatory to use the provided Kaggle JSON files dataset, or if there is flexibility to use another dataset that may better suit the requirements of the project? I am considering an alternative dataset that I believe could enhance the accuracy and efficiency of the gesture recognition model I am developing. I look forward to your guidance on this matter. Thank you for your time and assistance. Best regards, Sai Bhaskar Kandula, Ph:+91 9930623923 Github:SaiBhaskar0987

Go ahead with the alternative dataset that you found out. Can you please share the source/URL of the dataset you are planning to work with?

**Comment by SaiBhaskar0987** [comment]
TQ sir, I will let you know about dataset in 2 days sir, I need to research on it.

**Comment by bhoomika0910** [comment]
Hi! I'm a GSSoC '26 contributor with hands-on 
experience in Python, Deep Learning, and Computer 
Vision (currently building a YOLOv8-based assistive 
system). I'd love to work on this project and submit 
a well-documented Jupyter Notebook with model 
training, evaluation, and results.

Could you please assign this to me? 🙏

**Comment by abhisheks008** [comment]
> Hi! I'm a GSSoC '26 contributor with hands-on experience in Python, Deep Learning, and Computer Vision (currently building a YOLOv8-based assistive system). I'd love to work on this project and submit a well-documented Jupyter Notebook with model training, evaluation, and results.
> 
> Could you please assign this to me? 🙏

Share your detailed appproach with 4 models.
